File: server/src/main/java/com/github/genraven/gradlejavaserver/domain/Difficulty.java
package com.github.genraven.gradlejavaserver.domain;

import com.fasterxml.jackson.annotation.JsonValue;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum Difficulty {
    EASY("Easy"),
    AVERAGE("Average"),
    HARD("Hard"),
    DAUNTING("Daunting"),
    FORMIDABLE("Formidable");

    @JsonValue
    private final String label;
}

File: client/src/models/common/Difficulty.ts
import {Option} from "../../components/common/InputSelectField";

export enum Difficulty {
    Easy = 'Easy',
    Average = 'Average',
    Hard = 'Hard',
    Daunting = 'Daunting',
    Formidable = 'Formidable'
}

export const getDifficultyDice = (difficulty: Difficulty) => {
    switch (difficulty) {
        case Difficulty.Easy:
            return 1;
        case Difficulty.Average:
            return 2;
        case Difficulty.Hard:
            return 3;
        case Difficulty.Daunting:
            return 4;
        case Difficulty.Formidable:
            return 5;

    }
}

export const getDifficultyOptions = (): Option[] => {
    return Object.values(Difficulty).map((value) => ({value}))
}